File: Leetcode/Practice/Array/2419. Longest Subarray With Maximum Bitwise AND.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans = 0, c = 0;

        for(int i = 0; i < nums.size(); i++) {
            int j = i, cc = 0;
            while(j < nums.size() and nums[j] == nums[i]) j++, cc++;
            if(nums[i] > ans) ans = nums[i], c = cc;
            else if(nums[i] == ans) c = max(c, cc);
            i = j-1;
        }

        return c;
    }
};

File: Leetcode/Practice/Array/2419. Longest Subarray With Maximum Bitwise AND.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        mx = max(nums)
        ans = c = 0
        
        for num in nums:
            if num == mx:
                c += 1
            else:
                c = 0
            ans = max(ans, c)
        
        return ans